driver/src/main/scala/core/nodeset.scala

```diff
@@ -663,16 +667,20 @@ final class ChannelFactory private[reactivemongo] (
     pipeline
   }
 
-  private def sslContext = {
+  private def sslContext(tm: Array[javax.net.ssl.TrustManager]) = {
+    import java.io.FileInputStream
+    import java.security.{ KeyStore, Security }
+    import javax.net.ssl.KeyManagerFactory
 
-    val keyManagers: Array[KeyManager] = Option(System.getProperty("javax.net.ssl.keyStore")).map { path =>
+    val keyManagers = Option(System.getProperty("javax.net.ssl.keyStore")).map { path =>
 
       val password = Option(System.getProperty("javax.net.ssl.keyStorePassword")).getOrElse("")
 
       val ks = {
         val res = KeyStore.getInstance("JKS")
         val fin = new FileInputStream(path)
         res.load(fin, password.toCharArray)
+        fin.close()
```

Comment:
ok it is now, I was being a bit cheeky/cowboyish and assuming that you would only hit an IOException in the event of misconfiguration and then your not going to be running the application for long anyway as it won't work. But I've put the close in a finally block now, thought about doing something fancier such as a scala function implementing try with resources but decided to keep it simple.